src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/PKCS1MaskGenerationMethod.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    public class PKCS1MaskGenerationMethod : MaskGenerationMethod
+    {
+        private string _hashNameValue;
+        
+        public PKCS1MaskGenerationMethod()
+        {
+            _hashNameValue = "SHA1";
+        }
+
+        public string HashName
+        {
+            get { return _hashNameValue; }
+            set 
+            { 
+                if (value != null && value != "SHA1")
+                {
+                    throw new PlatformNotSupportedException();
+                }
+
+                _hashNameValue = "SHA1";
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA5350", Justification = "CryptoConfig.CreateFromName is not supported on Unix platforms, should create SHA1 directly")]
```

Comment:
CryptoConfig.CreateFromName("SHA1") returns a type that currently only works on Windows.  So the message is a bit incorrect/overreaching.

I'd say "CryptoConfig.CreateFromName may return platform-dependent objects, so this implementation is limited to SHA-1 for now", or something like that.